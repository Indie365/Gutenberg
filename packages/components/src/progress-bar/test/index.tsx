/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { ProgressBar, INDETERMINATE_TRACK_WIDTH } from '..';

describe( 'ProgressBar', () => {
	it( 'should render an indeterminate semantic progress bar element', () => {
		render( <ProgressBar /> );

		const progressBar = screen.getByRole( 'progressbar' );

		expect( progressBar ).toBeVisible();
		expect( progressBar ).not.toHaveValue();
	} );

	it( 'should render a determinate semantic progress bar element', () => {
		render( <ProgressBar value={ 55 } /> );

		const progressBar = screen.getByRole( 'progressbar' );

		expect( progressBar ).toBeVisible();
		expect( progressBar ).toHaveValue( 55 );
	} );

	it( 'should use `INDETERMINATE_TRACK_WIDTH`% as track width for indeterminate progress bar', () => {
		const { container } = render( <ProgressBar /> );

		/**
		 * We're intentionally not using an accessible selector, because
		 * the track is an intentionally non-interactive presentation element.
		 */
		// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
		const indicator = container.querySelector(
			'.components-progress-bar__indicator'
		);

		expect( indicator ).toHaveStyle( {
			width: `${ INDETERMINATE_TRACK_WIDTH }%`,
		} );
	} );

	it( 'should use `value`% as width for determinate progress bar', () => {
		const { container } = render( <ProgressBar value={ 55 } /> );

		/**
		 * We're intentionally not using an accessible selector, because
		 * the track is an intentionally non-interactive presentation element.
		 */
		// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
		const indicator = container.querySelector(
			'.components-progress-bar__indicator'
		);

		expect( indicator ).toHaveStyle( {
			width: '55%',
		} );
	} );

	it( 'should have an `is-indeterminate` className if `value` is not provided', () => {
		const { container } = render( <ProgressBar /> );

		expect( container ).toMatchSnapshot();
	} );

	it( 'should not have an `is-indeterminate` className if `value` is provided', () => {
		const { container } = render( <ProgressBar value={ 55 } /> );

		expect( container ).toMatchSnapshot();
	} );

	it( 'should be able to specify custom track and indicator colors', () => {
		const { container } = render(
			<ProgressBar indicatorColor="#900" trackColor="#d9d9d9" />
		);

		expect( container ).toMatchSnapshot();
	} );
} );
