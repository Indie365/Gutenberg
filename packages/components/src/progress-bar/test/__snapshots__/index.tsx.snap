// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should be able to specify custom track and indicator colors 1`] = `
<div>
  <div
    class="components-progress-bar is-indeterminate"
    style="background-color: rgb(217, 217, 217);"
  >
    <div
      class="components-progress-bar__indicator"
      style="width: 50%; background-color: rgb(153, 0, 0);"
    />
    <progress
      max="100"
    />
  </div>
</div>
`;

exports[`ProgressBar should have an \`is-indeterminate\` className if \`value\` is not provided 1`] = `
<div>
  <div
    class="components-progress-bar is-indeterminate"
  >
    <div
      class="components-progress-bar__indicator"
      style="width: 50%;"
    />
    <progress
      max="100"
    />
  </div>
</div>
`;

exports[`ProgressBar should not have an \`is-indeterminate\` className if \`value\` is provided 1`] = `
<div>
  <div
    class="components-progress-bar"
  >
    <div
      class="components-progress-bar__indicator"
      style="width: 55%;"
    />
    <progress
      max="100"
      value="55"
    />
  </div>
</div>
`;
